pkg/endpoint/policy.go

```diff
@@ -714,6 +724,17 @@ func (e *Endpoint) Regenerate(regenMetadata *regeneration.ExternalRegenerationMe
 
 	regenContext := ParseExternalRegenerationMetadata(ctx, cFunc, regenMetadata)
 
+	// Compute policy on the first regeneration before queueing the regeneration event
+	err := e.ComputeInitialPolicy(regenContext)
+	if err != nil {
+		cFunc()
+		if !errors.Is(err, context.Canceled) {
+			e.getLogger().WithError(err).Error("Initial policy compute failed")
+		}
+		close(done)
+		return done
+	}
+
```

Comment:
Why put this code here and not in the first piece of code that is executed when the regen event is dequeued?

Reply:
Good question, I've assumed there is some sort of rate limiting on the dequeue end, perhaps?

Reply:
Maybe this is better placed in `EndpointRegenerationEvent.Handle()`, as then this would parallelize better...

Reply:
As long as this is done prior to `e.owner.QueueEndpointBuild()` which acquires a "build permit".

Reply:
Moved this to `EndpointRegenerationEvent.Handle()` and clarified locking requirements (policy compute needs the endpoint `buildMutex`). `Handle` also serializes the build per endpoint, so new race conditions are less likely.